### PR TITLE
Reflow some code samples.

### PR DESCRIPTION
**Pull request by drwells**
The content of @code/@endcode blocks must be less than 80 characters per line to prevent clang-format from rewrapping it in strange ways.

I went ahead and fixed three files: things look better now.

Part of #6814.

### CODE REVIEW

File: include/deal.II/base/array_view.h [hunk 1]
```diff
@@ -48,15 +48,19 @@ DEAL_II_NAMESPACE_OPEN
...
+ * std::vector<int> array = get_data();
+ *
+ * // a view of elements 5..9 (inclusive)
+ * ArrayView<int> view (&array[5], 5);
+ * view[2] = 42; // array[7] is set to 42
+ *
+ * // same view, but read-only
+ * ArrayView<const int> const_view (&array[5], 5);
+ * // returns 42
+ * int element_7 = const_view[2];
```
Comment (bangerth) [inline include/deal.II/base/array_view.h]: can you attach the `returns 42` to the line in question? it's not clear the way it looks right now whether that's for the following or previous line.

**Comment by drwells** [comment]
Lets not merge these until #6822 is decided.

File: include/deal.II/base/array_view.h (hunk 1 above)
Comment (masterleinad) [reply include/deal.II/base/array_view.h]: @drwells ping

Comment (bangerth) [reply include/deal.II/base/array_view.h]: Or maybe better change the comment to "set element_7 to 42", but indeed after the statement in question

**Comment by drwells** [comment]
@bangerth @masterleinad I updated the documentation in `array_view.h`; I put all comments on the ends of the related lines of code. I think this is the least ambiguous option.

**Pull request merged**